src/bin/main.rs

```diff
@@ -28,13 +28,36 @@ use url::Url;
 
 #[tokio::main]
 async fn main() {
+    let mut args = std::env::args();
```

Comment:
In the future we should use `clap` here instead of parsing from args .. it creates a very ergonomic cli from the binary https://docs.rs/clap/latest/clap/

Reply:
If it is OK with you, I would introduce `clap` using the derive pattern instead of the builder pattern.  The latter has runtime failures, while the former has compile time failures which I find much better for rapid development.

Reply:
+1 on that. compile time failures are better than runtime failures